BuildConfigGen/Program.cs

```diff
@@ -1032,6 +1071,62 @@ private static void WriteTaskJson(string taskPath, TaskStateStruct taskState, Co
             ensureUpdateModeVerifier!.WriteAllText(outputTaskPath, outputTaskNode.ToJsonString(jso), suppressValidationErrorIfTargetPathDoesntExist: false);
         }
 
+        /// <summary>
+        /// This uses the same major.minor.patch for all build configuration tasks, but the "build" suffix of semver is different, and directly corresponds to the name.
+        /// We no longer populate the '_buildConfigMapping' property of the task.json, since server won't expect this property to be set.
+        /// </summary>
+        /// <param name="taskPath"></param>
+        /// <param name="taskState"></param>
+        /// <param name="defaultConfig"></param>
+        /// <param name="config"></param>
+        /// <param name="fileName"></param>
+        /// <param name="existingLocalPackageVersion"></param>
+        private static void WriteTaskJsonWithSemverConfig(string taskPath,
```

Comment:
Better to combine this with WriteTaskJson, and just add an if statement to opt into "withSemverConfig")